docs/content/guides/developer/app-examples/trustless-swap.mdx

```diff
@@ -1048,741 +1275,1284 @@ Note that `dApp Kit` provides a set of hooks for making query and mutation calls
 
-The main mutating logic is fairly straightforward, executing a Move call of a package named `demo_bear::new` to create a dummy bear object and transfer it to the connected wallet account, all within the same `Transaction`. The example reuses the `executeTransaction()` hook from the **Execute Transaction Hook** step to execute the transaction.
+The main mutating logic includes executing a Move call of a package named `demo_bear::new` to create a dummy bear object and transfering it to the connected wallet account, all within the same `Transaction`. The example reuses the `executeTransaction()` hook from the **Execute Transaction Hook** step to execute the transaction.
 
 Another benefit of wrapping the main mutating logic inside `useMutation()` is that you can access and manipulate the cache storing server state. The example fetches the cache from remote servers by using query call in an appropriate callback. In this case, it is the `onSuccess` callback. When the transaction succeeds, invalidate the cache data at the cache key called `getOwnedObjects`, then `@tanstack/react-query` handles the re-fetching mechanism for the invalidated data automatically. Do this by using `invalidateQueries()` on the `@tanstack/react-query` configured client instance retrieved by `useQueryClient()` hook in the **Setup Providers** step.
 
 Now the logic to create a dummy bear object exists. You just need to attach it into the button in the header.
 
-```ts title='src/components/Header.tsx'
-import { useGenerateDemoData } from '@/mutations/demo';
+<details>
+  <summary>
+    `Header.tsx`
+  </summary>
+  {@inject: examples/trading/frontend/src/components/Header.tsx}
+</details>
 
-export function Header() {
-	const { mutate: demoBearMutation, isPending } = useGenerateDemoData();
-	return (
-		<Container>
-			<Box>
-				<Button
-					className="cursor-pointer"
-					disabled={isPending}
-					onClick={() => {
-						demoBearMutation();
-					}}
-				>
-					New Demo Bear
-				</Button>
-			</Box>
-		</Container>
-	);
-}
-```
+:::danger[Checkpoint]
 
-### Lock/unlock owned-object mutation
+Run your app and ensure you can:
 
-Locking and unlocking of owned objects are two crucial on-chain actions in this application and are very likely to be used all over. Hence, it's beneficial to extract their logic into separated mutating functions to enhance reusability and encapsulation.
+- Mint a demo bear object and view it in the `Manage Objects` tab.
 
-#### Lock owned objects
+:::
+
+### Locking owned objects
+
+Locking owned objects is a crucial on-chain action in this application and is very likely to be used all over. Hence, it's beneficial to extract its logic into separated mutating functions to enhance reusability and encapsulation.
```

Comment:
I don't think the reason we extract this out is because it's crucial operation, or even that it's likely to be re-used. @manolisliolios can clarify, but I think it's:

- For encapsulation -- so the complexity for the query is hidden from the component that uses it.
- For discoverability -- so that if someone wanted to find a particular mutation, they know they can go the `mutations` sub-directory and start their search from there.